aptos-move/framework/aptos-framework/sources/ManagedCoin.move

```diff
@@ -0,0 +1,151 @@
+
+    /// When no capabilities (burn/mint) found on an account.
+    const ENO_CAPABILITIES: u64 = 0;
+
+    // Data structures
+
+    /// Capabilities resource storing mint and burn capabilities.
+    /// The resource is stored on the account that initialized coin `CoinType`.
+    struct Capabilities<phantom CoinType> has key {
+        mint_cap: MintCapability<CoinType>,
+        burn_cap: BurnCapability<CoinType>,
+    }
+
+    // Public functions
+
+    /// Withdraw an `amount` of coin `CoinType` from `account` and burn it.
+    public(script) fun burn<CoinType>(
+        account: &signer,
+        amount: u64,
+    ) acquires Capabilities {
+        let type_info = TypeInfo::type_of<CoinType>();
+        let coin_address = TypeInfo::account_address(&type_info);
+
+        assert!(
+            exists<Capabilities<CoinType>>(coin_address),
+            Errors::not_published(ENO_CAPABILITIES),
+        );
+
+        let capabilities = borrow_global<Capabilities<CoinType>>(coin_address);
+
+        let to_burn = Coin::withdraw<CoinType>(account, amount);
+        Coin::burn(to_burn, &capabilities.burn_cap);
+    }
+
+    /// Initialize new coin `CoinType` in Aptos Blockchain.
+    /// Mint and Burn Capabilities will be stored under `account` in `Capabilities` resource.
+    public(script) fun initialize<CoinType>(
+        account: &signer,
+        name: vector<u8>,
+        scaling_factor: u64,
+        monitor_supply: bool,
+    ) {
+        let (mint_cap, burn_cap) = Coin::initialize<CoinType>(
+            account,
+            ASCII::string(name),
+            scaling_factor,
+            monitor_supply,
+        );
+
+        move_to(account, Capabilities<CoinType>{
+            mint_cap,
+            burn_cap,
+        });
+    }
+
+    /// Create new coins `CoinType' and deposit them into dst_addr's account.
+    public(script) fun mint<CoinType>(
+        account: &signer,
+        dst_addr: address,
+        amount: u64,
+    ) acquires Capabilities {
+        let account_addr = Signer::address_of(account);
+
+        assert!(
+            exists<Capabilities<CoinType>>(account_addr),
+            Errors::not_published(ENO_CAPABILITIES),
+        );
+
+        let capabilities = borrow_global<Capabilities<CoinType>>(account_addr);
+        let coins_minted = Coin::mint(amount, &capabilities.mint_cap);
+        Coin::deposit(dst_addr, coins_minted);
+    }
+
+    /// Creating a resource that stores balance of `CoinType` on user's account, withdraw and deposit event handlers.
+    /// Required if user wants to start accepting deposits of `CoinType` in his account.
+    public(script) fun register<CoinType>(account: &signer) {
+        Coin::register<CoinType>(account);
+    }
+
+     /// Transfers `amount` of coins `CoinType` from `from` to `to`.
+    public(script) fun transfer<CoinType>(
+        from: &signer,
+        to: address,
+        amount: u64,
+    ) {
+         Coin::transfer<CoinType>(from, to, amount);
+    }
```

Comment:
Would prefer to not re-export these and instead use the common ones. That way wallets don't need to specialize.

Reply:
Good